### PR TITLE
Forms shared generate assets

### PR DESCRIPTION
**Pull request by MarekBodingerBA**
This script generates base64-encoded assets from files and strings and writes them to the src/generated-assets directory as TypeScript files.

After examining multiple solutions that:
  - load the file as a string (file loaders in esbuild/vite/webpack, using fs directly) - for static assets,
  - pre-evaluate script (babel-plugin-preval, esbuild custom plugins) - for scripts that need to access the file system
    in their evaluation - e.g. generating Tailwind or Inter font CSS

this no-frills approach was chosen for its simplicity and ease of use. It works in all environments (jest, Next.js/Nest) or built/directly imported version of the library consistently and is as performant as other solutions.

The generated file looks like this:
image

### CODE REVIEW

File: forms-shared/.gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 /dist
 .env
+# These assets are generated by "npm run generate-assets" and should not be committed
+/src/generated-assets
```
[the remark this answers is not in the document]
Comment (MarekBodingerBA) [reply forms-shared/.gitignore:4]: I don't know, they are directly imported ([like here](https://github.com/bratislava/konto.bratislava.sk/pull/1117/files#diff-1882239801a4df2d20e9d6012c405ced38db54b974d5a8dcd7d5e24c13f6b584)), and they are not only prerequisite for build, but also for development (running tests).

I don't think there is a good solution, these are indeed "src" files, but need to be pregenerated unfortunately.

[the remark this answers is not in the document]
Comment (MarekBodingerBA) [reply forms-shared/.gitignore:4]: One possibility is also to commit them, and stop doing "generate-assets" on postinstall. Which might be better when I think about it now.

**Review by mpinter — APPROVED** [review]
Ok, makes sense. LGTM

File: forms-shared/.gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (mpinter) [reply forms-shared/.gitignore]: :+1:, perhaps a quick note on "what kind of files are "assets" and what I need to do if I add those such as these" in the readme to accompany this ?

File: forms-shared/scripts/assets/assets.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+function convertStringToBase64(str: string) {
+  return Buffer.from(str).toString('base64')
+}
+
+const rootDir = path.join(__dirname, '../../')
+
+const assetsList = [
+  {
+    name: 'taxPdf',
```
[the remark this answers is not in the document]
Comment (MarekBodingerBA) [reply forms-shared/scripts/assets/assets.ts:29]: Not possible, look at `summaryPdfCss` which is code that needs to be ran.

File: forms-shared/.gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarekBodingerBA) [reply forms-shared/.gitignore]: I added it to my README todo.

**Pull request merged**